XiEditor/EditViewController.swift

```diff
@@ -516,10 +516,19 @@ class EditViewController: NSViewController, EditViewDataSource, FindDelegate, Sc
             mouseDragged(with: event)
         }
     }
-    
-    // used mostly for paste
+
+    // NOTE: this was previously used for paste, and could possibly be removed, but it
+    // may be used for IME?
     override func insertText(_ insertString: Any) {
-        document.sendRpcAsync("insert", params: insertedStringToJson(insertString as! NSString))
+        let text: String
+        if let s = insertString as? NSString {
+            text = s as String
+        } else if let s = insertString as? NSAttributedString {
```

Comment:
Ditto here.

Reply:
Hmm, I don't think it works here; I need access to the `.string` method on `NSAttributedString`, I don't think I can just cast an `NSAttributedString` to a `String`.